test/conformance/testing/include/uur/assert.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2022-2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#ifndef UUR_CONFORMANCE_TESTING_ASSERT_H_INCLUDED
+#define UUR_CONFORMANCE_TESTING_ASSERT_H_INCLUDED
+
+#include <cstdio>
+#include <cstdlib>
+
+#define UUR_ASSERT(CONDITION, MESSAGE)                                     \
+    if (!(CONDITION)) {                                                    \
+        std::fprintf(stderr, "%s: %d: %s\n", __FILE__, __LINE__, MESSAGE); \
+        std::abort();                                                      \
```

Comment:
I am just curious, does gtest catch the `SIGABRT` by default and declare the test as failed? (just concerned if that may terminate the whole tests if it fails or may not terminate the test without calling some necessary destructors like static variables if any)

Reply:
Neither of these macros seem to be used in this PR, maybe it shouldn't be added?

Reply:
I guess they were used inside `test/conformance/testing/source/utils.cpp`

Reply:
Oh, I was searching on a partial diff because GitHub is super helpful and doesn't load the whole thing... Disregard my comment then.

Reply:
> I am just curious, does gtest catch the `SIGABRT` by default and declare the test as failed? (just concerned if that may terminate the whole tests if it fails or may not terminate the test without calling some necessary destructors like static variables if any)

I tested this locally and it just aborts the entire process. No destructors are called. Maybe this is an issue, but this is how it was handled in UnitCL

Reply:
Yeah, this could be a real pain for a partially implemented adapter hitting aborts because a query failed. I'd be much better to propogate the error code and fail a fixture/test than abort.

Reply:
Its not as nice as `std::expected` but we should probably always return a `ur_result_t` and get the value as an out param.